File: backend/grafioschtrader-common/src/main/java/grafioschtrader/reportviews/currencypair/CurrencypairWithHistoryquote.java
package grafioschtrader.reportviews.currencypair;

import grafioschtrader.dto.ISecuritycurrencyIdDateClose;
import grafioschtrader.entities.Currencypair;

/**
 * It may be used to return the quotation for a Currencypair with a certain Date.
 */
public class CurrencypairWithHistoryquote {

  public Currencypair currencypair;
  public ISecuritycurrencyIdDateClose historyquote;

  public CurrencypairWithHistoryquote(Currencypair currencypair, ISecuritycurrencyIdDateClose historyquote) {
    this.currencypair = currencypair;
    this.historyquote = historyquote;
  }

}

File: frontend_old/src/app/entities/view/currencypair.with.historyquote.ts
import {Currencypair} from '../currencypair';
import {ISecuritycurrencyIdDateClose} from '../projection/i.securitycurrency.id.date.close';

export class CurrencypairWithHistoryquote {
  constructor(public currencypair: Currencypair, public historyquote: ISecuritycurrencyIdDateClose) {
  }
}
